src/Compilers/CSharp/Portable/Lowering/ExtensionMethodReferenceRewriter.cs

```diff
@@ -194,8 +194,7 @@ method.OriginalDefinition is ErrorMethodSymbol ||
 
         public override BoundNode? VisitMethodDefIndex(BoundMethodDefIndex node)
         {
-            MethodSymbol method = node.Method;
-            Debug.Assert(method.IsDefinition); // Tracked by https://github.com/dotnet/roslyn/issues/78962 : From the code coverage and other instrumentations perspective, should we remap the index to the implementation symbol? 
+            MethodSymbol method = VisitMethodSymbolWithExtensionRewrite(this, node.Method);
```

Comment:
:memo: we presently have no scenario where a `BoundMethodDefIndex` to an extension method would be created and used outside of the extension body. That's handled by `ExtensionMethodBodyRewriter`. So the corresponding change here cannot be observed at the moment.